### PR TITLE
Proposed bugfix for pull-to-refresh bug

### PR DESCRIPTION
**Pull request by namolnad**
It seems as though, when self.mediaImages is nil (after having deleted all objects), asking for the minID parameter results in the failed initialization of the parameters NSDictionary. Then when calling the populateDataWithParameters method, it is expecting an NSDictionary, but getting nil instead. I decided to initialize an empty dictionary and then add the min_id parameter dictionary if self.mediaItems.count is >= 1.

### CODE REVIEW

File: Blocstagram/BLCDataSource.m [hunk 1]
```diff
@@ -82,11 +82,14 @@ - (void) populateDataWithParameters:(NSDictionary *)parameters completionHandler
                            
                            NSMutableString *urlString = [NSMutableString stringWithFormat:@"https://api.instagram.com/v1/users/self/feed?access_token=%@", self.accessToken];
                            
-                           for (NSString *paramterName in parameters)
-                           {
-                               // For example, if dictionary contains {count: 50}, append `&count=50' to the URL
-                               [urlString appendFormat:@"&%@=%@", paramterName, parameters[paramterName]];
-                           }
+//                           if (![parameters isEqualToDictionary:@{}])
```
Comment (mhanlon) [inline Blocstagram/BLCDataSource.m]: Nice. It's a little thing, but tidying up details like this will help in the long run.

**Comment by mhanlon** [comment]
Great approach.

**Comment by namolnad** [comment]
Perfect. @mhanlon, should I merge this into master? Perhaps after uncommenting the code ' if (![parameters isEqualToDictionary:@{}])'

Do you think it's also worth it to approach the requestOldItemsWithCompletionHandler in the same way as the requestNewItems method? That is, initializing the dictionary prior to setting it equal to maxID? Though I don't think the same problem exists—since it's impossible to scroll down if there are no mediaItems—I'd still imagine it'd be good practice to do so. Thoughts?

**Comment by mhanlon** [comment]
I think the `requestOldItemsWithCompletionHandler` is fine, because the issue wasn't so much passing a `nil` dictionary as it was trying to add a `nil` value to the dictionary... There's no harm in passing in an empty dictionary, but nil is just as valid.

Merge it back to master, for sure.

**Pull request merged**